src/Plugin/InstallTask/SiteSettings.php

```diff
@@ -0,0 +1,215 @@
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('http_client'),
+      $container->get('externalauth.authmap'),
+      $container->get('logger.factory'),
+      $container->get('state')
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entityTypeManager, ClientInterface $client, AuthmapInterface $authmap, LoggerChannelFactoryInterface $logger_factory, StateInterface $state) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->client = $client;
+    $this->authmap = $authmap;
+    $this->logger = $logger_factory->get('stanford_profile');
+    $this->state = $state;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function runTask(array &$install_state) {
+    // @codeCoverageIgnoreStart
+    if (!static::isAhEnv()) {
+      return;
+    }
+    // @codeCoverageIgnoreEnd
+    $site_name = $install_state['forms']['install_configure_form']['site_name'] ?? self::DEFAULT_SITE;
+    $site_name = Html::escape($site_name);
+
+    $site_data = $this->getSnowData($site_name);
+    if (empty($site_data)) {
+      return;
+    }
+    $this->entityTypeManager->getStorage('config_pages')->create([
+      'type' => 'stanford_basic_site_settings',
+      'su_site_email' => $site_data['email'],
+      'su_site_name' => $site_data['webSiteTitle'],
+      'context' => 'a:0:{}',
+    ])->save();
+
+    $this->addSiteOwner($site_data['sunetId'], $site_data['email']);
+
+    if (isset($site_data['webSiteOwners'])) {
+      foreach ($site_data['webSiteOwners'] as $owner) {
+        if ($owner['sunetId'] == $site_data['sunetId']) {
+          continue;
+        }
+
+        $this->addSiteOwner($owner['sunetId'], $owner['email']);
+      }
+    }
+  }
+
+  /**
+   * Add a user with the site manager role.
+   *
+   * @param string $sunet
+   *   User SunetId.
+   * @param string $email
+   *   User email.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function addSiteOwner($sunet, $email) {
+    $new_user = $this->entityTypeManager->getStorage('user')->create([
+      'name' => $sunet,
+      'pass' => user_password(),
+      'mail' => $email,
+      'roles' => ['site_manager'],
+      'status' => 1,
+    ]);
+    $new_user->save();
+    $this->authmap->save($new_user, 'simplesamlphp_auth', $sunet);
+  }
+
+  /**
+   * Get site information from the SNOW API.
+   *
+   * @param string $site_name
+   *   The requested name of the site.
+   *
+   * @return array|null
+   *   Returned data if any exist.
+   */
+  protected function getSnowData($site_name) {
+    $api_url = $this->state->get('stanford_profile_snow_api', self::SNOW_API);
```

Comment:
How would one change the state before getting to this if it is run during install time as the final step?

Reply:
hmm. i wonder if this can be set in a settings.php type of thing

Reply:
I don't think you can set state in settings.php. I'm pretty sure it is only config.

Reply:
yeah i cant find anything about overriding or providing state in settings. boooo

Reply:
unless we have a switch-a-roo in here. :man_shrugging:

Reply:
:man_facepalming: I'll change it to settings. not state. then we can set the setting in post settings hook

Reply:
this should be better now. and should be identical to the D7 approach with the url and credentials
```
$settings['stanford_profile_snow_api_url'] = 'whatever';
$settings['stanford_profile_snow_api_user'] = 'whatever_username';
$settings['stanford_profile_snow_api_pass'] = 'whatever_password';
```